src/file_upload.rs

```diff
@@ -101,7 +101,14 @@ fn handle_multipart(
 /// invalid.
 /// This method returns future.
 pub fn upload_file(req: &HttpRequest<crate::MiniserveConfig>) -> FutureResponse<HttpResponse> {
-    let app_root_dir = req.state().path.clone().canonicalize().unwrap();
+    let return_path: String = req.headers()[header::REFERER]
+        .to_str()
+        .unwrap_or("/")
+        .to_owned();
+    let app_root_dir = match req.state().path.canonicalize() {
+        Ok(path) => path,
+        Err(_) => return Box::new(create_error_response("Internal server error", &return_path)),
```

Comment:
Can't you get rid of the `match` here as you don't care about the `Err` anyway? You could use `let app_root_dir = if let ...`.